source/MaterialXShaderGen/ShaderGenerators/Glsl/OgsFx/OgsFxShaderGenerator.cpp

```diff
@@ -15,20 +15,62 @@ namespace
         { "i_normal", "NORMAL" },
         { "i_tangent", "TANGENT" },
         { "i_bitangent", "BITANGENT" },
-        { "i_texcoord0", "TEXCOORD0" },
-        { "i_texcoord1", "TEXCOORD1" },
-        { "i_texcoord2", "TEXCOORD2" },
-        { "i_texcoord3", "TEXCOORD3" },
-        { "i_texcoord4", "TEXCOORD4" },
-        { "i_texcoord5", "TEXCOORD5" },
-        { "i_texcoord6", "TEXCOORD6" },
-        { "i_texcoord7", "TEXCOORD7" },
-        { "i_texcoord8", "TEXCOORD8" },
-        { "i_texcoord9", "TEXCOORD9" },
+
+        { "i_texcoord_0", "TEXCOORD0" },
+        { "i_texcoord_1", "TEXCOORD1" },
+        { "i_texcoord_2", "TEXCOORD2" },
+        { "i_texcoord_3", "TEXCOORD3" },
+        { "i_texcoord_4", "TEXCOORD4" },
+        { "i_texcoord_5", "TEXCOORD5" },
+        { "i_texcoord_6", "TEXCOORD6" },
+        { "i_texcoord_7", "TEXCOORD7" },
+        { "i_texcoord_8", "TEXCOORD8" },
+        { "i_texcoord_9", "TEXCOORD9" },
+
+        { "i_color3_0", "COLOR0" },
```

Comment:
I'd need to double check but I don't think you can have more than 2 color units.

Reply:
Ah! Just let me know if that's the case and I will simplify this.

Reply:
This is what's supported in OGSFX so only 1 color input. Last number is slots used so we actually only have 6 tex coord slots without clashing with bitangent+tangent.

POSITION position 0
NORMAL normal 1
COLOR0 color 2
BINORMAL binormal 14
TANGENT tangent 13
TEXCOORD0-7 texcoord 0-7 7 - 14

Reply:
Ok, I'll change that.